common/api/core-backend.api.md

```diff
@@ -5175,7 +5175,7 @@ export class SqliteChangesetReader implements IDisposable {
     } & SqliteChangesetReaderArgs): SqliteChangesetReader;
     static openGroup(args: {
         readonly changesetFiles: string[];
-    } & SqliteChangesetReaderArgs): SqliteChangesetReader;
+    } & MarkRequired<SqliteChangesetReaderArgs, "db">): SqliteChangesetReader;
```

Comment:
Do I need to deprecate the old version of this to make this change? @iTwin/itwinjs-core-admins

Reply:
Technically no since the class is `@beta` but for something like this, I'd defer to someone who knows this particular code better. (Was this a mistake? Is there a valid case for not passing a db in the args?)

Reply:
@nick4598

Reply:
This is a brand new API I believe (around a month old).

From [SQLite Docs on changegroups](https://www.sqlite.org/session/sqlite3changegroup_add.html):

>If the new changeset contains changes to a table that is already present in the changegroup, then the number of columns and the position of the primary key columns for the table must be consistent. If this is not the case, this function fails with SQLITE_SCHEMA. Except, if the changegroup object has been configured with a database schema using the sqlite3changegroup_schema() API, then it is possible to combine changesets with different numbers of columns for a single table, provided that they are otherwise compatible.
 
 It is possible to use the changegroup API with or without a database provided, but I don't know how practical it would be to not provide a database in the context of iTwin.js. We should expect in pretty much every case that we atleast have one to pass... but which version of the database should be passed? 
 
 The way the changegroup API resolves the differences requires that there exists a database table with:
- the table name identified by the changeset, 
- at least as many columns as recorded in the changeset, and
- the primary key columns in the same position as recorded in the changeset.

Since we want the database table to have more columns than our changeset, I assume we'd want to pass the database which has the schema already applied to it, just incase that schema added a new column. (What if a schema removes a column? Does this happen? I don't know)
 
 @khanaffan might be worried about how the changegroup API currently resolves the differences? A changeset table with fewer columns than the database schema would just have the default value used for the missing columns. I don't know if that would be a problem or not.

Reply:
So  `SqliteChangesetReader` does not require db context.  If found it useful in debugging some changeset where all i wanted to is to e.g. print out table/pk/op.  Grouping require a db context due to schema changes. And ECChangeReader definitely require db context.

We can simplify this by making db a required argument. The debug case is not really important. But there is still no control on what version the db is on. May be added docs there that say it should be a head of change set been open.

Also, openLoalChanges() also take iModel from which it open local changes. Maybe we would not need the iModel arg if we make db arg non-optional

`public static openLocalChanges(args: { iModel: IModelJsNative.DgnDb, includeInMemoryChanges?: true } & SqliteChangesetReaderArgs): SqliteChangesetReader`


The api is beta so we can make this change.

Reply:
@nick4598 we do not really have default values all new columns are default to null at least in ecdb/imodel

Reply:
I've made `db` required in `SqliteChangesetReaderArgs` and never undefined in `SqliteChangesetReader`.

openLocalChanges() only works with an `IModelDb` database, not an `ECDb` database (even looking down into the C++ code, that's what is required). So only for that function, I decided to keep the `args` parameter since all of the args are used, but I forced the type of `SqliteChangesetReaderArgs.db` to be `IModelDb`.